### PR TITLE
Clean up dependencies

### PR DESCRIPTION
**Pull request by calebmshafer**
Remove a handful of unneeded dependencies

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,25 +43,12 @@
...
-    "eslint": "^7.31.0",
-    "eslint-plugin-react-hooks": "^4.2.0",
-    "fibers": "^5.0.0",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
     "react-redux": "^7.2.0",
-    "redux": "^4.1.1",
-    "type-fest": "^0.13.1",
-    "utf-8-validate": "^5.0.5"
+    "redux": "^4.1.1"
```
Comment (tcobbs-bentley) [inline package.json:49]: @calebmshafer I notice that you removed redux from mobile-ui-react, and you are adding it here. Does mobile-sdk-core use redux?

Comment (tcobbs-bentley) [reply package.json:50]: Oops, I misread; you aren't adding it, but leaving it. The question remains: is it needed?

Comment (calebmshafer) [reply package.json:50]: Good question, I don't think it does.  We can probably remove redux and react-redux.

Comment (tcobbs-bentley) [reply package.json:50]: I think all react packages should be removed from here.

Comment (calebmshafer) [reply package.json:50]: Plus the only reason I left the React dependency is due to the ui-core dependency which I'd definitely want to re-visit when moving to 3.0.

The only use is UiEvent which I don't even understand why it exists over just the BeUiEvent in @bentley/bentleyjs-core.

Comment (tcobbs-bentley) [reply package.json:50]: I'll approve this PR, and look at the rest.

**Review by tcobbs-bentley — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (calebmshafer) [reply package.json:50]: sounds good. Thanks!

**Pull request merged**